aiida/storage/psql_dos/migrator.py

```diff
@@ -312,6 +306,27 @@ def initialise_database(self) -> None:
             context.stamp(context.script, 'main@head')
             self.connection.commit()
 
+    def delete_tables(self, exclude_tables: list[str] | None = None) -> None:
```

Comment:
Maybe this could be called `delete_all_tables` ?

Otherwise `migrator.delete_tables(['table1','table2'])` can be misleading.

You could, on top, consider enforcing `exclude_tables` to be keyword-only by something like `def delete_all_tables(self,*,exclude_tables: list[str] | None = None) -> None:`